src_c/polygon.c

```diff
@@ -437,6 +437,148 @@ pg_polygon_get_vertices(pgPolygonObject *self, void *closure)
+        PyErr_SetString(PyExc_TypeError, "Must assign numeric values");
+        return -1;
+    }
+    poly->vertices[i * 2] = Vx;
+    poly->vertices[i * 2 + 1] = Vy;
+    return 0;
+}
+
+static Py_ssize_t
+pg_polygon_seq_length(pgPolygonObject *self)
+{
+    return pgPolygon_GETVERTSNUM(self);
+}
+
+static int
+pg_polygon_ass_subscript(pgPolygonObject *self, PyObject *op, PyObject *value)
+{
+    if (PyIndex_Check(op)) {
+        PyObject *index;
+        Py_ssize_t i;
+
+        index = PyNumber_Index(op);
+        if (index == NULL) {
+            return -1;
+        }
+        i = PyNumber_AsSsize_t(index, NULL);
+        Py_DECREF(index);
+        return pg_polygon_ass_vertex(self, i, value);
+    }
+    else {
+        PyErr_SetString(PyExc_TypeError, "Expected a number or sequence");
+        return -1;
+    }
+}
+
+static PyObject *
+pg_polygon_subscript(pgPolygonObject *self, PyObject *op)
+{
+    PyObject *index = PyNumber_Index(op);
+    Py_ssize_t i;
+
+    if (index == NULL) {
+        return NULL;
+    }
+    i = PyNumber_AsSsize_t(index, NULL);
+    Py_DECREF(index);
+
+    pgPolygonBase *poly = &self->polygon;
+
+    if (i < 0) {
+        i += poly->verts_num;
+    }
+
+    if (i < 0 || i > poly->verts_num) {
+        return RAISE(PyExc_IndexError, "Invalid vertex Index");
+    }
+
+    return pg_TupleFromDoublePair(poly->vertices[i * 2], poly->vertices[i * 2 + 1]);
+}
+
+static int
+pg_polygon_contains_seq(pgPolygonObject *self, PyObject *arg)
+{
+    double x, y;
+    if (!PySequence_FAST_CHECK(arg)) {
+        PyErr_SetString(PyExc_TypeError, "Expected a sequence");
+        return -1;
+    }
+    if (!pg_TwoDoublesFromObj(arg, &x, &y)) {
+        PyErr_SetString(PyExc_TypeError, "Expected a sequence of 2 numbers");
+        return -1;
+    }
+    else if (PySequence_Check(arg)) {
+        if (PySequence_Size(arg) < 2) {
+            PyErr_SetString(PyExc_TypeError, "Expected a sequence of 2 numbers");
+            return -1;
+        }
+        PyObject *tmp = PySequence_GetItem(arg, 0);
+        if (!tmp) {
+            return -1;
+        }
+        
+        if (!pg_DoubleFromObj(tmp, &x)) {
+            PyErr_SetString(PyExc_TypeError, "Expected a sequence of 2 numbers");
+            return -1;
+        }
+        Py_DECREF(tmp);
+
+        tmp = PySequence_GetItem(arg, 1);
+        if (!tmp) {
+            return NULL;
+        }
+        
+        if (!pg_DoubleFromObj(tmp, &y)) {
+            PyErr_SetString(PyExc_TypeError, "Expected a sequence of 2 numbers");
+            return -1;
+        }
+        Py_DECREF(tmp);
+
+        pgPolygonBase *poly = &self->polygon;
+        Py_ssize_t i;
+        for (i = 0; i < poly->verts_num; i+=2) {
+            if (poly->vertices[i] == x && poly->vertices[i+1] == y) {
+                return 1;
+            }
+        }
+    }
+    else {
+        PyErr_SetString(PyExc_TypeError, "Expected a sequence");
+        return -1;
+    }
+    return 0;
+}
```

Comment:
This function can be massively simplified when #100 will be merged.

Reply:
And it's flawed already, at first you're discarding sequences that are not tuples or lists, but later you check for sequences.

Reply:
When #100 is merged we could have something as simple as this:
```C
static int
pg_polygon_contains_seq(pgPolygonObject *self, PyObject *arg)
{
    double x, y;
    
    if (!pg_TwoDoublesFromObj(arg, &x, &y)) {
        PyErr_SetString(PyExc_TypeError, "Expected a sequence of 2 numbers");
        return -1;
    }
    
    pgPolygonBase *poly = &self->polygon;
    Py_ssize_t i;
    for (i = 0; i < poly->verts_num; i+=2) {
        if (poly->vertices[i] == x && poly->vertices[i+1] == y) {
            return 1;
        }
    }
    
    return 0;
}
```